src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -435,18 +442,48 @@ public void Clear()
             }
         }
 
-        // Having lots of extra slime jets and mucocysts really doesn't help you that much
+        // Having lots of extra pili, slime jets and mucocysts doesn't really help much
+        pilusScore *= MathF.Sqrt(pilusCount);
         slimeJetScore *= MathF.Sqrt(slimeJetsCount);
-        slimeJetScore *= slimeJetsMultiplier;
-
         mucocystsScore *= MathF.Sqrt(mucocystsCount);
 
+        slimeJetScore *= slimeJetsMultiplier;
+
         var predationToolsRawScores = (pilusScore, oxytoxyScore, slimeJetScore, mucocystsScore);
 
         cachedPredationToolsRawScores.Add(microbeSpecies, predationToolsRawScores);
         return predationToolsRawScores;
     }
 
+    public (float EnzymesScore, bool IsMembranedigestable) GetEnzymesScores(MicrobeSpecies predator,
+        string dissolverEnzyme)
+    {
+        var key = (predator, dissolverEnzyme);
+        if (cachedEnzymeScores.TryGetValue(key, out var cached))
+            return cached;
+
+        var organelles = predator.Organelles.Organelles;
+        var isMembraneDigestable = dissolverEnzyme == Constants.LIPASE_ENZYME;
+        var enzymesScore = 1.0f;
+
+        foreach (var organelle in organelles)
+        {
+            if (organelle.Definition.HasLysosomeComponent)
+            {
+                foreach (var enzyme in organelle.Definition.Enzymes.Where(enzyme => enzyme.Value > 0))
```

Comment:
This LINQ usage should be avoided as this allocates a bunch of temporary memory. I'm doing some style changes already for this PR, so I can change this as well. I'll just comment for letting you know to watch out for this in the future.

Reply:
Also I think I found a major problem here, using `Name` instead of `InternalName` means that the scores were *never* read from the score dictionary. I'll push what I have but I think I'll need your help to rebalance all the numbers to work.